### ISSUE TITLE
Use default path setting in MSYS2

### ISSUE THREAD

**Issue by renatosilva** [appveyor/ci#813]
Feodor, can you please revert the Windows path inheritance performed in #785? MSYS2 stopped doing this by default, so MSYS2_PATH_TYPE has been introduced to restore previous behavior. However you are doing this for all projects, instead of each project setting the variable as needed.

That issue would be better reported to the MSYS2 project instead of AppVeyor, if they want to change default path type. Alternatively, they can use the p7zip package and take advantage of new behavior to fix [their own problems](https://github.com/JuliaLang/julia/blob/f020d38/appveyor.yml#L44-L49) with external path.

**Comment by FeodorFitsner** [comment]
But you can reset this variable in your project, no?

While crawling the internet for the solution I got the impression that for many people having the same PATH is more natural - they use bash/shell as a replacement for Windows batch and they want to be able to run the same tools as in cmd session.

**Comment by renatosilva** [comment]
The point is making a default MSYS2 installation for running clean builds. The same way you install only basic packages, you would allow for MSYS2 to decide default settings instead of making it unpredictable. @Alexpux says that this change was due to build problems with external tools. Whatever is preferable, the point is allowing MSYS2 to handle this.

As for that specific report, the switch to minimal path actually uncovered a problem with Julia itself, where they assume 7-Zip is available instead of installing the dependency, and automatically fixed two other as mentioned. Maybe the inherited path is good for daily usage but not for neutral environments.

**Comment by FeodorFitsner** [comment]
Adding @tkelman to the conversation.

**Comment by tkelman** [comment]
Taking windows tools and things users add to their windows path off of it by default is a huge mistake by msys2. Cygwin would never have done such a thing as a default setting. If you want an isolated environment for building msys2 packages, handle it in pkgbuild, like homebrew does. Don't prevent users from running applications where they can't use msys2's version (does the msys2 python work with virtualenv yet?) or where msys2 will never have a version (powershell, signtool, etc).

**Comment by renatosilva** [comment]
If you are unsatisfied with the change, even though it fixes problems in your own project, please report this to MSYS2 and convince Alexey to restore previous behavior, not AppVeyor to tweak the installation with unpredictable customization.

**Comment by tkelman** [comment]
AppVeyor inherited a major breaking change by upgrading msys2, I complained because this broke my appveyor builds that had previously been working, and Feodor promptly put in place a mitigation to fix the breakage. I don't use msys2 except on appveyor since caching a cygwin build would be more time consuming to set up on every build. If msys2 is going to make disruptive breaking changes like that I doubt I'm going to be able to convince them to do otherwise or walk back on that decision.

Feodor has a responsibility to his paying customers to avoid causing major breaking changes to existing working build configurations. That is in conflict with your desire for appveyor to use the default configuration of msys2. The default should probably be less breaking. I have little confidence in msys2's maintainers to respond to external feedback considering how issues and pull requests I've made in the past have sat permanently unaddressed.

**Comment by renatosilva** [comment]
So the project leader decided something that benefits your project by fixing two problems and uncovering another, you then disagree that this was good, refuse to argue with upstream, refuse to make a simple variable setting, and comes here complain because you pay? Does not sound reasonable.

**Comment by tkelman** [comment]
I don't think I'm the only for whom this change causes issues. I tracked the upstream change down to https://github.com/Alexpux/MSYS2-packages/commit/6e0e609da2efce404695cdceb99215dcc03769f7, which looks like it was just pushed to master without any discussion or feedback? Was there a mailing list thread or issue or anything?

I'm not going to wait for msys2 to maybe fix something and have broken builds until then. Setting an environment variable would probably have worked too, but Feodor found it before I did and having changed it for all AppVeyor users causes less breakage than forcing everyone to do it manually in every single project that needs it.

Using the system copy of 7zip is not a "problem" that was uncovered, it was a desirable feature, I don't want to make every appveyor build slower and more dependent on msys2 servers by installing the msys2 7z every time. Last I checked it's also way out of date. The other things I am manually removing from path are problems that were already fixed.
